### ISSUE TITLE
[Feature] [runtime env] [java] select jdk version

### ISSUE THREAD

**Issue by SongGuyang** [ray-project/ray#21239]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

In some real world cases, users need to use different jdk versions to run their jobs in one cluster. We can support this feature in runtime env.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by jovany-wang** [comment]
I believe it's a strong requirement. But I have concerns on what APIs we'd like to expose.
1. Should we expose the API to specify a JDK version for actor or task? or we only need to expose it for job.
2. Once different workers are ran in different JDK versions, we have to make sure the compatibilities with different versions in our Ray Java codebase.

Hence, maybe we should only expose the API for job level. 2 options:
1. JDK versions as enum to choose.
```yaml
ray {
  job {
    "jdk-version": "JDK11"
  }  
}
```
2.  Setting $JAVA_HOME
```yaml
ray {
  job {
    "JAVA_HOME": "/usr/local/jdk11"
  }  
}
```

Also CC @chaokunyang @kfstorm @raulchen

**Comment by raulchen** [comment]
I'm in favor of the second option. Because 1) the concrete install path is managed by the users, not by Ray. So Ray doesn't know where "JDK11" is. 2) There are too many JDK versions, Oracle, OpenJDK, etc. It's hard for Ray to manage all of them.

**Comment by raulchen** [comment]
@SongGuyang Python should support different versions as well. Is this requirement tracked by any other issues?

**Comment by SongGuyang** [comment]
> @SongGuyang Python should support different versions as well. Is this requirement tracked by any other issues?

No issue now. In python, maybe user could use conda ?

**Comment by raulchen** [comment]
Right, Conda can solve this issue as well. But I'm not sure if Conda is too heavy-weighted for some users. Anyway, we can support Conda first, and see if there will be requirements for using different Python interpretors without Conda.

**Comment by jovany-wang** [comment]
Let me give the conclusion on the jdk selection:
1. Users could configure the jdk home for a cluster, like:
```bash
ray start --jdks={"jdk11": "/path/to/jdk11/home", "my_jdk": "/path/to/myjdk/home"}
```

2. Then users could specify the jdk in the runtime env.
```java
RuntimeEnv env = new RuntimeEnv.Builder().setJDK("my_jdk");
// use the env and the Java worker will be use the my_jdk for the worker.
```

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
